numpy/core/src/umath/reduction.c

```diff
@@ -304,11 +301,14 @@ PyUFunc_ReduceWrapper(PyArrayObject *operand, PyArrayObject *out,
 
     result = NpyIter_GetOperandArray(iter)[0];
 
+    int needs_api = NpyIter_IterationNeedsAPI(iter);
```

Comment:
You want this before clearing the exception flags?

Reply:
It doesn't matter, but it fit better with what is coming, I think (because `needs_api` needs to be updated also by the loop fetching, which has to happen before the float clearing).

I guess its a bit tricky about what to do with `PyArray_CopyInitialReduceValues` and float status.